x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/filters/filters.tsx

```diff
@@ -0,0 +1,348 @@
+  setFilters: Function;
+  indexPattern: IndexPattern;
+  defaultQuery: Filter;
+}) => {
+  const [isOpenByCreation, setIsOpenByCreation] = useState(false);
+  const [localFilters, setLocalFilters] = useState(() =>
+    filters.map((filter) => ({ ...filter, id: generateId() }))
+  );
+
+  const updateFilters = (updatedFilters: FilterValue[]) => {
+    // do not set internal id parameter into saved object
+    setFilters(updatedFilters.map((filter) => omit(filter, 'id')));
+    setLocalFilters(updatedFilters);
+  };
+
+  const onAddFilter = () =>
+    updateFilters([
+      ...localFilters,
+      {
+        ...defaultQuery,
+        id: generateId(),
+      },
+    ]);
+  const onRemoveFilter = (id: string) =>
+    updateFilters(localFilters.filter((filter) => filter.id !== id));
+
+  const onChangeValue = (id: string, query: Query, label: string) =>
+    updateFilters(
+      localFilters.map((filter) =>
+        filter.id === id
+          ? {
+              ...filter,
+              input: query,
+              label,
+            }
+          : filter
+      )
+    );
+
+  const onDragEnd = ({
+    source,
+    destination,
+  }: {
+    source?: DraggableLocation;
+    destination?: DraggableLocation;
+  }) => {
+    if (source && destination) {
+      const items = euiDragDropReorder(localFilters, source.index, destination.index);
+      updateFilters(items);
+    }
+  };
+
+  return (
+    <>
+      <EuiDragDropContext onDragEnd={onDragEnd}>
+        <EuiDroppable droppableId="FILTERS_DROPPABLE_AREA" spacing="s">
+          {localFilters?.map((filter: FilterValue, idx: number) => {
+            const { input, label, id } = filter;
+            const queryIsValid = isQueryValid(input, indexPattern);
+
+            return (
+              <EuiDraggable
+                spacing="m"
+                key={id}
+                index={idx}
+                draggableId={id}
+                disableInteractiveElementBlocking
+              >
+                {(provided) => (
+                  <EuiPanel paddingSize="none">
+                    <EuiFlexGroup gutterSize="s" alignItems="center" responsive={false}>
+                      <EuiFlexItem grow={false}>{/* Empty for spacing */}</EuiFlexItem>
+                      <EuiFlexItem grow={false}>
+                        <EuiIcon
+                          size="s"
+                          color={queryIsValid ? 'subdued' : 'danger'}
+                          type={queryIsValid ? 'grab' : 'alert'}
+                          title={
+                            queryIsValid
+                              ? i18n.translate('xpack.lens.indexPattern.filters.dragToReorder', {
+                                  defaultMessage: 'Drag to reorder',
+                                })
+                              : i18n.translate('xpack.lens.indexPattern.filters.isInvalid', {
+                                  defaultMessage: 'This query is invalid',
+                                })
```

Comment:
Nice to have, but not necessary, would be an indication of why it's invalid.

Reply:
We're not able to extract this information :(